packages/tools/src/stateManagement/segmentation/polySeg/polySegConverters.js

```diff
@@ -0,0 +1,387 @@
+    imageData.setSpacing(spacing);
+
+    const scalarArray = vtkDataArray.newInstance({
+      name: 'Pixels',
+      numberOfComponents: 1,
+      values: scalarData,
+    });
+
+    imageData.getPointData().setScalars(scalarArray);
+
+    imageData.modified();
+
+    for (const index of segmentIndices) {
+      const annotations = annotationUIDsInSegmentMap.get(index);
+
+      for (const annotation of annotations) {
+        const bounds = getBoundingBoxAroundShapeWorld(
+          annotation.data.contour.polyline
+        );
+
+        const [iMin, jMin, kMin] = utilities.transformWorldToIndex(imageData, [
+          bounds[0][0],
+          bounds[1][0],
+          bounds[2][0],
+        ]);
+
+        const [iMax, jMax, kMax] = utilities.transformWorldToIndex(imageData, [
+          bounds[0][1],
+          bounds[1][1],
+          bounds[2][1],
+        ]);
+
+        // Run the pointInShapeCallback for the combined bounding box
+        pointInShapeCallback(
+          imageData,
+          (pointLPS) => {
+            // Check if the point is inside any of the polylines for this segment
+            return isPointInsidePolyline3D(
+              pointLPS,
+              annotation.data.contour.polyline
+            );
+          },
+          ({ pointIJK }) => {
+            segmentationVoxelManager.setAtIJKPoint(pointIJK, index);
+          },
+          [
+            [iMin, iMax],
+            [jMin, jMax],
+            [kMin, kMax],
+          ]
+        );
+      }
+    }
+
+    return segmentationVoxelManager.scalarData;
+  },
+  async convertContourToStackLabelmap(args, ...callbacks) {
+    const [progressCallback] = callbacks;
+    const polySeg = await new ICRPolySeg();
+    await polySeg.initialize({
+      updateProgress: progressCallback,
+    });
+
+    const { segmentationsInfo, annotationUIDsInSegmentMap, segmentIndices } =
+      args;
+
+    const segmentationVoxelManagers = new Map();
+
+    segmentationsInfo.forEach((segmentationInfo, referencedImageId) => {
+      const { dimensions, scalarData, direction, spacing, origin } =
+        segmentationInfo;
+      const manager = utilities.VoxelManager.createVolumeVoxelManager(
+        dimensions,
+        scalarData
+      );
+
+      const imageData = vtkImageData.newInstance();
+      imageData.setDimensions(dimensions);
+      imageData.setOrigin(origin);
+      imageData.setDirection(direction);
+      imageData.setSpacing(spacing);
+
+      const scalarArray = vtkDataArray.newInstance({
+        name: 'Pixels',
+        numberOfComponents: 1,
+        values: scalarData,
+      });
+
+      imageData.getPointData().setScalars(scalarArray);
+
+      imageData.modified();
+
+      segmentationVoxelManagers.set(referencedImageId, { manager, imageData });
+    });
+
+    for (const index of segmentIndices) {
+      const annotations = annotationUIDsInSegmentMap.get(index);
+
+      for (const annotation of annotations) {
+        if (!annotation.data) {
```

Comment:
I get an undefined occasionally for annotation - not that annotation.data isn't defined, but that annotation itself is undefined.

Reply:
I think I saw that sometime (that is why i have a debugger) but tried it again and can't reproduce it at all